common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -1307,6 +1340,14 @@ private void initialize23(@NonNull final KeyPairGenerator keyPairGenerator,
         keyPairGenerator.initialize(spec);
     }
 
+    @SuppressLint("NewApi")
+    @RequiresApi(Build.VERSION_CODES.N)
+    @NonNull
+    private KeyGenParameterSpec.Builder setAttestationChallenge(
```

Comment:
Can we add a test for these?

Reply:
The `getCertificateChain()` is the test -- [that function won't return a result unless an attestation challenge is set](https://developer.android.com/reference/android/security/keystore/KeyGenParameterSpec.Builder#setAttestationChallenge(byte[])) -- if there's some other aspect of this we can test, I'm open to ideas. There didn't seem to be a lot to do here

Reply:
Yeah , this is private.  Oops.